tests/integration/test_kms.py

```diff
@@ -239,6 +240,48 @@ def test_generate_data_key_pair(self, kms_client, kms_key):
         )
         assert decrypted["Plaintext"] == result["PrivateKeyPlaintext"]
 
+    @pytest.mark.parametrize("number_of_bytes", [12, 44, 91, 1, 1024])
+    @pytest.mark.aws_validated
+    def test_generate_random(self, kms_client, snapshot, number_of_bytes):
+        result = kms_client.generate_random(NumberOfBytes=number_of_bytes)
+
+        plain_text = result.get("Plaintext")
+        assert plain_text
+        assert isinstance(plain_text, bytes)
+        assert len(plain_text) == number_of_bytes
+        snapshot.match("result_length", len(plain_text))
+
+    @pytest.mark.parametrize(
+        "number_of_bytes,error_type,error_message",
+        [
+            (
+                None,
+                botocore.exceptions.ClientError,
+                "NumberOfBytes is required.",
+            ),
+            (
+                0,
+                botocore.exceptions.ClientError,
+                "Member must have value greater than or equal to 1",
+            ),
+            (
+                1025,
+                botocore.exceptions.ClientError,
+                "Member must have value less than or equal to 1024",
+            ),
+        ],
```

Comment:
The exception type and message should be stored and asserted by the snapshot utility. No reason to specify this here! Parametrizing only the number_of_bytes argument will suffice.